simulators/solar_attenuator/__init__.py

```diff
@@ -0,0 +1,76 @@
+# Author:
+#   Lorenzo Monti <lorenzo.monti@inaf.it>
+from socketserver import ThreadingTCPServer
+from simulators.common import ListeningSystem
+
+servers = [
+    (('0.0.0.0', 13100), (), ThreadingTCPServer, {}),
+]
+
+
+class System(ListeningSystem):
+
+    commands = {
+        'set W_solar_attn': 'set_w_solar_attn',
+        'set W_cal': 'set_w_cal',
+        'set W_passthrough': 'set_w_passthrough',
+        'get W_cal_temp': 'get_w_cal_temp',
+        'get W_mode': 'get_w_mod',
+    }
+
+    tail = '\r\n'
+
+    def __init__(self):
+        self.mode = ""
+        self.cal_temp = 21.0
+        self._set_default()
+
+    def _set_default(self):
+        self.msg = ''
+
+    def parse(self, byte):
+        if byte == '\n':  # Ending char
+            msg = self.msg
+            self._set_default()
+            return self._parse(msg)
+        else:
+            self.msg += byte
+            return True
+
+    def _parse(self, msg):
+        commandList = msg.split(';')
+        answer = ''
+        for command in commandList:
+            args = command.split()
+            if len(args) < 2:
+                continue
+            cmd_name = self.commands.get(args[0] + " " + args[1])
+            if not cmd_name:
+                continue
```

Comment:
I believe this line is not covered by tests, lowering overall coverage to 99.x%. @LorenzoMonti can you write a test to cover this line as well?

Reply:
@giuseppe-carboni done!